pkg/algorithm/nodeselect/v1alpha1/select_node.go

```diff
@@ -163,9 +206,7 @@ func (ac *Config) selectNode(nodeBlockDeviceMap map[string]*blockDeviceList) *no
 // diskFilterConstraint takes a value for key "ndm.io/disk-type" and form a label.
 func diskFilterConstraint(diskType string) string {
 	var label string
-	if diskType == string(apis.TypeSparseCPV) {
-		label = string(apis.NdmDiskTypeCPK) + "=" + string(apis.TypeSparseCPV)
-	} else {
+	if blockdevice.SupportedDiskType[diskType] {
 		label = string(apis.NdmBlockDeviceTypeCPK) + "=" + string(apis.TypeBlockDeviceCPV)
```

Comment:
we need to consider for sparse as well

Reply:
From blockdevice we have only one label `ndm.io/blockdevice-type: blockdevice`
```yaml
 kubectl get bd sparse-5a92ced3e2ee21eac7b930f670b5eab5 -n openebs -o yaml
apiVersion: openebs.io/v1alpha1
kind: BlockDevice
metadata:
  creationTimestamp: "2019-06-11T18:51:05Z"
  generation: 1
  labels:
    kubernetes.io/hostname: 127.0.0.1
    ndm.io/blockdevice-type: blockdevice
  name: sparse-5a92ced3e2ee21eac7b930f670b5eab5
  namespace: openebs
  resourceVersion: "1077"
  selfLink: /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/sparse-5a92ced3e2ee21eac7b930f670b5eab5
  uid: ddba8cb6-8c79-11e9-b2a4-54e1ad4a9dd4
spec:
  capacity:
    logicalSectorSize: 0
    physicalSectorSize: 0
    storage: 10737418240
  details:
    compliance: ""
    deviceType: sparse
    firmwareRevision: ""
    model: ""
    serial: ""
    vendor: ""
  devlinks: []
  filesystem: {}
  partitioned: "No"
  path: /var/openebs/sparse/0-ndm-sparse.img
status:
  claimState: Unclaimed
  state: Active
```
so no need to consider separate case for sparse